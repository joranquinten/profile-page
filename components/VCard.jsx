import React from 'react';
import styled from 'styled-components';
import renderHTML from 'react-render-html';

import { Row, Col } from 'antd';
import SVGIcon from './SVGIcon';

import { footer, contact } from '../data/nl';

const StyledVCard = styled.div`
  width: 100%;
`;

const Iconic = styled.div`
  font-size: 10px;
  display: flex;
  align-items: flex-start;

  svg {
    height: 24px;
  }

  div,
  span {
    font-size: 1.4rem;
  }
`;

const VCard = () => (
  <StyledVCard id="hcard-Joran-Quinten" className="vcard">
    <Row gutter={16}>
      <Col span={8}>
        <Iconic>
          <SVGIcon icon="coffee" />
          {renderHTML(footer.description)}
        </Iconic>
      </Col>
      <Col span={8}>
        <Iconic>
          <SVGIcon icon="guy" />
          <div>
            <a className="url fn" href={contact.personalUrl}>
              {contact.fullName}
            </a>
            <br />
            <a className="url org" href={contact.organisationUrl}>
              {contact.organisation}
            </a>
            <br />
            <a className="email" href={`mailto:${contact.email}`}>
              {contact.email}
            </a>
            <br />
            <span className="tel">{contact.phone}</span>
          </div>
        </Iconic>
      </Col>
      <Col span={8}>
        <Iconic>
          <SVGIcon icon="paperPlane" />
          <span className="adr">
            <span className="street-address">{contact.address.street}</span>
            <br />
            <span className="postal-code">{contact.address.zipcode}</span>
            <br />
            <span className="locality">{contact.address.city}</span>
          </span>
        </Iconic>
      </Col>
    </Row>
  </StyledVCard>
);

export default VCard;
