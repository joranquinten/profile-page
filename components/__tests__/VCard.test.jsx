import { shallow } from 'enzyme';

import VCard from '../VCard';

jest.mock('../../data/nl.js');

describe('<VCard />', () => {
  it('should render without crashing', () => {
    expect(shallow(<VCard />)).toMatchSnapshot();
  });
});
