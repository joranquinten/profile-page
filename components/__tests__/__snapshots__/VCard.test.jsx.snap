// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VCard /> should render without crashing 1`] = `
<VCard__StyledVCard
  className="vcard"
  id="hcard-Joran-Quinten"
>
  <Row
    gutter={16}
  >
    <Col
      span={8}
    >
      <VCard__Iconic>
        <Icon
          icon="coffee"
        />
        <span
          key="0"
        >
          footerDescription
        </span>
      </VCard__Iconic>
    </Col>
    <Col
      span={8}
    >
      <VCard__Iconic>
        <Icon
          icon="guy"
        />
        <div>
          <a
            className="url fn"
            href="contactPersonalUrl"
          >
            contactFullName
          </a>
          <br />
          <a
            className="url org"
            href="contactOrganisationUrl"
          >
            contactOrganisation
          </a>
          <br />
          <a
            className="email"
            href="mailto:contactEmail"
          >
            contactEmail
          </a>
          <br />
          <span
            className="tel"
          >
            contactPhone
          </span>
        </div>
      </VCard__Iconic>
    </Col>
    <Col
      span={8}
    >
      <VCard__Iconic>
        <Icon
          icon="paperPlane"
        />
        <span
          className="adr"
        >
          <span
            className="street-address"
          >
            contactStreet
          </span>
          <br />
          <span
            className="postal-code"
          >
            contactZipcode
          </span>
          <br />
          <span
            className="locality"
          >
            contactCity
          </span>
        </span>
      </VCard__Iconic>
    </Col>
  </Row>
</VCard__StyledVCard>
`;
